test-clients/src/main/java/com/hedera/services/bdd/suites/contract/openzeppelin/ERC1155ContractInteractions.java

```diff
@@ -0,0 +1,154 @@
+	private static final Logger log = LogManager.getLogger(ERC1155ContractInteractions.class);
+	private static final String CONTRACT_NAME = "ERC-1155";
+	private static final String OPERATIONS_PAYER = "payer";
+	private static final String ACCOUNT1 = "acc1";
+	private static final String ACCOUNT2 = "acc2";
+	private static final String CONTRACT_FILE_NAME = "contractFileName";
+	public static void main(String... args) {
+		new ERC1155ContractInteractions().runSuiteSync();
+	}
+	
+	private ByteString getFileContents(String path) {
+		ByteString contents = null;
+		try {
+			var bytes = Files.readAllBytes(Path.of(path));
+			contents = ByteString.copyFrom(bytes);
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		return contents;
+	}
+	
+	@Override
+	protected List<HapiApiSpec> getSpecsInSuite() {
+		return List.of(
+				erc1155()
+		);
+	}
+
+	private HapiApiSpec erc1155() {
+		final var contents = getFileContents(ContractResources.ERC_1155_BYTECODE_PATH);
+		return defaultHapiSpec(CONTRACT_NAME)
+				.given(
+						cryptoCreate(ACCOUNT1),
+						cryptoCreate(ACCOUNT2),
+						cryptoCreate(OPERATIONS_PAYER).balance(ONE_MILLION_HBARS),
+						fileCreate(CONTRACT_FILE_NAME).payingWith(OPERATIONS_PAYER),
+						updateLargeFile(OPERATIONS_PAYER, CONTRACT_FILE_NAME, contents)
+				)
+				.when()
+				.then(
+						contractCreate(CONTRACT_NAME).bytecode(CONTRACT_FILE_NAME).via("contractCreate").payingWith(OPERATIONS_PAYER),
+						getTxnRecord("contractCreate").logged(), // 121618 gas
+						getAccountBalance(OPERATIONS_PAYER).logged(), // started with 1M hbars
+						getAccountInfo(ACCOUNT1).savingSnapshot(ACCOUNT1 + "Info"),
+						getAccountInfo(ACCOUNT2).savingSnapshot(ACCOUNT2 + "Info"),
+						getAccountInfo(OPERATIONS_PAYER).savingSnapshot(OPERATIONS_PAYER + "Info"),
+						withOpContext((spec, log) -> {
+							var accountOneAddress = spec.registry().getAccountInfo(ACCOUNT1 + "Info").getContractAccountID();
+							var accountTwoAddress = spec.registry().getAccountInfo(ACCOUNT2 + "Info").getContractAccountID();
+							var operationsPayerAddress = spec.registry().getAccountInfo(OPERATIONS_PAYER + "Info").getContractAccountID();
+							var ops = new ArrayList<HapiSpecOperation>();
+
+							/* approve for other accounts */
+							var approveCall1 = contractCall(CONTRACT_NAME, ContractResources.ERC_1155_ABI_APPROVE, accountOneAddress, true)
+									.via("acc1ApproveCall").payingWith(OPERATIONS_PAYER)
+									.hasKnownStatus(ResponseCodeEnum.SUCCESS);
+							ops.add(approveCall1);
+							var approveCall2 = contractCall(CONTRACT_NAME, ContractResources.ERC_1155_ABI_APPROVE, accountTwoAddress, true)
+									.via("acc2ApproveCall").payingWith(OPERATIONS_PAYER)
+									.hasKnownStatus(ResponseCodeEnum.SUCCESS);
+							ops.add(approveCall2);
+
+							/* mint to some account */
+							var mintCall = contractCall(CONTRACT_NAME, ContractResources.ERC_1155_ABI_MINT, 0, 10, accountOneAddress)
```

Comment:
You need to mint only to the `operations_payer` so that you are able to transfer from his NFTS to the accountOne account